codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServiceConfigGenerator.kt

```diff
@@ -104,15 +105,42 @@ data class Initializer(
  * }
  *
  * has two variable bindings. The `bar` name is bound to a `String` variable and the `baz` name is bound to a
- * `u64` variable.
+ * `u64` variable. Both are bindings that use concrete types. Types can also be generic:
+ *
+ * ```rust
+ * fn foo<T>(bar: T) { }
  * ```
  */
-data class Binding(
-    /** The name of the variable. */
-    val name: String,
-    /** The type of the variable. */
-    val ty: RuntimeType,
-)
+sealed class Binding {
+    data class Generic(
```

Comment:
I tend to avoid using inheritance if I can. This pattern allows us to have [Rust-like enums in Kotlin](https://blog.shadura.me/2019/09/25/kotlin-enums/), which I prefer and I've seen in the codebase.